### PR TITLE
Recipe shortcode: on AMP, change the Print link

### PR DESCRIPTION
**Pull request by kienstra**
* For the [Recipe shortcode](https://wordpress.com/support/recipes/) in AMP, changes the 'Print' link to 'Print page,' and it now prints the entire page 
* No intended change to non-AMP
* In AMP, it's not possible to only print only the recipe, like on non-AMP
* Printing only the recipe is [done with a jQuery plugin](https://href.li/?https://github.com/Automattic/jetpack/blob/feeadafa22309b7cf0d77077c0354d9df1f74eda/modules/shortcodes/js/recipes-printthis.js).
* Even `<amp-script>` doesn't look possible here, as it [doesn't allow](https://amp.dev/documentation/components/amp-script/#creating-amp-elements) creating an `<iframe>`

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This enhances an existing feature
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here:
pbIAo1-1m-p2

#### Testing instructions:
1. Ensure the [AMP plugin](https://wordpress.org/plugins/amp/) is active
2. In AMP > Settings, select Transitional mode
3. In Jetpack > Settings > Writing, ensure this is toggled on:

settings-writing-sharing

4. Create a new post
5. Add a shortcode block with this, taken from the [documentation](https://wordpress.com/support/recipes/)
```html
[recipe title="Summer Pasta with Basil, Tomatoes and Cheese" servings="4-6" preptime="20 mins" cooktime="10 mins" difficulty="easy" rating="★★★★★" image="https://en-support.files.wordpress.com/2014/01/food-dinner-pasta-broccoli.jpg" description="A fresh, light, Italian-inspired pasta recipe perfect for a late summer dinner."]
[recipe-notes]
Credit: allrecipes.com
[/recipe-notes][recipe-ingredients]
– 2 pounds vine ripened tomatoes
– 3 cloves garlic
– 1/2 cup chopped fresh basil
[/recipe-ingredients][recipe-directions]
1. In a medium bowl, toss together tomatoes, garlic, basil, mint, salt and black pepper.
2. Cook and drain pasta.
3. Mix together.
[/recipe-directions][recipe-nutrition] Per Serving: 580 calories; 33 g fat; 50.6 g carbohydrates; 20.3 g protein; 50 mg cholesterol; 760 mg sodium. [/recipe-nutrition]
[/recipe]
```
6. Preview the non-AMP front-end
7. Preview the AMP front-end
8. Expected: the AMP front-end looks the same, except it has a 'Print page' link, not 'Print':

recipe-in-amp

9. Expected: clicking 'Print page' prints the entire page

#### Proposed changelog entry for your changes:

* Improve AMP compatibility of Recipe shortcode

### CODE REVIEW

**Comment by kienstra** [comment]
The failed unit tests don't look to be caused by this PR, though I could be wrong.

They also appear in the [latest commit](https://github.com/Automattic/jetpack/pull/15319/commits/9f449fc0517b03b55eb1420b90c908261ecfd821) for https://github.com/Automattic/jetpack/pull/15319.

File: tests/php/modules/shortcodes/test-class.recipe.php [hunk 1]
```diff
@@ -7,17 +7,72 @@
...
-		wp_reset_postdata();
+	/**
+	 * Tear down after each test.
+	 *
+	 * @inheritDoc
+	 */
+	public function tearDown() {
+		wp_dequeue_style( 'jetpack-recipes-js' );
+		wp_dequeue_script( 'jetpack-recipes-js' );
+		wp_dequeue_script( 'jetpack-recipes-printthis' );
```
Comment (kienstra) [inline tests/php/modules/shortcodes/test-class.recipe.php:35]: It's a little strange that these are needed, but the tests fail locally without them.

File: tests/php/modules/shortcodes/test-class.recipe.php [hunk 2]
```diff
@@ -7,17 +7,71 @@
...
-		// Reset data.
-		wp_reset_postdata();
+	/**
+	 * Tear down after each test.
+	 *
+	 * @inheritDoc
+	 */
+	public function tearDown() {
+		wp_dequeue_script( 'jetpack-recipes-js' );
+		wp_dequeue_script( 'jetpack-recipes-printthis' );
```
Comment (kienstra) [inline tests/php/modules/shortcodes/test-class.recipe.php:34]: It's a little strange that these are needed, but the tests fail locally without them.

**Comment by jeherve** [comment]
I rebased to address failing tests that should now be fixed.

**Comment by jeherve** [comment]
Synchronized with WordPress.com. Internal reference: r206183-wpcom

**Pull request merged**